Source/WebKit/UIProcess/API/APIContentRuleListStore.cpp

```diff
@@ -445,20 +449,29 @@ static WTF::String getContentRuleListSourceFromMappedFile(const MappedData& mapp
 {
     ASSERT(!RunLoop::isMain());
 
+    if (mappedData.metaData.version == std::numeric_limits<decltype(mappedData.metaData.version)>::max()) {
+        WTFLogAlways("Content Rule List source recovery failed: Version is invalid.");
+        return { };
+    }
+
     if (mappedData.metaData.version < 9) {
         WTFLogAlways("Content Rule List source recovery failed: Version is too old to recover the original JSON source from disk.");
         return { };
     }
 
     auto sourceSizeBytes = mappedData.metaData.sourceSize;
-    if (!sourceSizeBytes) {
+    if (!sourceSizeBytes || sourceSizeBytes == std::numeric_limits<decltype(sourceSizeBytes)>::max()) {
```

Comment:
These checks seem unnecessary.  Wouldn't the first check be hit?

Reply:
It could still overflow if the sizes are large but not max due to corruption.